streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -325,14 +325,15 @@ private void shouldImmediatelyAddStatelessTasksToRestoredTasks(final StreamTask.
     public void shouldRestoreSingleActiveStatefulTask() throws Exception {
         final StreamTask task =
             statefulTask(TASK_0_0, mkSet(TOPIC_PARTITION_A_0, TOPIC_PARTITION_B_0)).inState(State.RESTORING).build();
+        final AtomicBoolean allChangelogCompleted = new AtomicBoolean(false);
         when(changelogReader.completedChangelogs())
             .thenReturn(Collections.emptySet())
             .thenReturn(mkSet(TOPIC_PARTITION_A_0))
-            .thenReturn(mkSet(TOPIC_PARTITION_A_0, TOPIC_PARTITION_B_0));
-        when(changelogReader.allChangelogsCompleted())
-            .thenReturn(false)
-            .thenReturn(false)
-            .thenReturn(true);
+            .thenAnswer(invocation -> {
+                allChangelogCompleted.set(true);
+                return mkSet(TOPIC_PARTITION_A_0, TOPIC_PARTITION_B_0);
+            });
+        when(changelogReader.allChangelogsCompleted()).thenReturn(allChangelogCompleted.get());
```

Comment:
Would we need a "return always" or do we know `allChangelogsCompleted` is only called exactly ones? -- The original code indicates it would be called 3 times?

Reply:
The last `.thenReturn()` determines the return value for all following calls to `allChangelogCompleted()`.
However, I realize now that it should probably be
```
when(changelogReader.allChangelogsCompleted()).thenAnswer(invocation -> allChangelogCompleted.get());
```  
Otherwise, I would always return `false`. I am a bit confused now, because the test ran successfully.

I have to revisit this.